File: java/PE045.java
public class PE045 {
    public static void main(String[] args)
    {
        for(long i = 286; ; i++) {
            long m = T(i);
            if(isPentagonal(m) && isHexagonal(m)) {
                System.out.println(m);
                break;
            }
        }
    }

    private static long T(long n)
    {
        return n*(n+1)/2;
    }

    private static boolean isPentagonal(long n)
    {
        long t = (1+(long)Math.sqrt(1+24*n))/6;
        if(t*(3*t-1)/2 == n)
            return true;
        return false;
    }

    private static boolean isHexagonal(long n)
    {
        long t = (1+(long)Math.sqrt(1+8*n))/4;
        if(t*(2*t-1) == n)
            return true;
        return false;
    }
}

File: c/PE045.c
#include <stdio.h>
#include <math.h>

int isPentagonal(long long m)
{
    long long k = 1 + 24*m;
    int t = (int)(sqrt(k)+0.1);
    long long l = (1+t)/6;
    if((l*(3*l-1))/2 == m)
        return 1;
    return 0;
}
int isHexagonal(long long m){
    long long k = 1 + 8*m;
    int t = (int)(sqrt(k)+0.1);
    int l = (1+t)/4;
    if((l*(2*l-1)) == m)
        return 1;
    return 0;
}
int main()
{
    long long i;
    long long m;
    for(i = 286; ; i++){
        m = (i*(i+1))/2;
        if(isPentagonal(m) && isHexagonal(m))
            break;
    }
    printf("%lld\n", m);
    return 0;
}
